src/Lean/Elab/Syntax.lean

```diff
@@ -201,18 +201,20 @@ where
         throwError "unknown parser declaration/category/alias '{id}'"
 
   processSepBy (stx : Syntax) := do
+    let sepBy := mkCIdentFrom stx[0] ``ParserDescr.sepBy
     let p ← ensureUnaryOutput <$> withNestedParser do process stx[1]
     let sep := stx[3]
     let psep ← if stx[4].isNone then `(ParserDescr.symbol $sep) else ensureUnaryOutput <$> withNestedParser do process stx[4][1]
     let allowTrailingSep := !stx[5].isNone
-    return (← `(ParserDescr.sepBy $p $sep $psep $(quote allowTrailingSep)), 1)
+    return (← `($sepBy:ident $p $sep $psep $(quote allowTrailingSep)), 1)
```

Comment:
@Kha This doesn't work. (I mean, it doesn't hurt but it doesn't do anything.) I was hoping that by annotating the span on the `ParserDescr.sepBy` I would get hover info on `sepBy(` in syntax, but it doesn't seem to attach any info. Any idea why?

Reply:
This should add synthetic positions, which are ignored by the language server

Reply:
Untested, `(with_annotate_term $(stx[0]) ParserDescr.sepBy) ...` might work

Reply:
great, that worked. I pushed a simplification which avoids the need for `parser_alias%`, since it turns out that `toParserDescr` is actually in `TermElabM` rather than `IO` or `MacroM` like I originally thought, so you can just add terminfo directly.